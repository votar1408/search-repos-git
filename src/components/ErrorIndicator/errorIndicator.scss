.error-indicator {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.error-indicator span:nth-child(2) {
    font-size: 16px;
}
