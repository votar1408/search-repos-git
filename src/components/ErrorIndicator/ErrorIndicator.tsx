import './errorIndicator.scss';

import React, {FC} from 'react';

import {texts} from '../../consts';

export const ErrorIndicator: FC = () => {
    const {errorIndicator1, errorIndicator2, errorIndicator3} = texts;

    return (
        <div className="error-indicator">
            <span>{errorIndicator1}</span>
            <span>{errorIndicator2}</span>
            <span>{errorIndicator3}</span>
        </div>
    );
};
