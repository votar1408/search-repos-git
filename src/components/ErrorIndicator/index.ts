import {ErrorIndicator} from './ErrorIndicator';

export {ErrorIndicator};
