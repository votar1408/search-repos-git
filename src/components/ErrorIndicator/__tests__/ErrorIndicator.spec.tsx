import React from 'react';

import {
    shallow,
    ShallowWrapper
} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {ErrorIndicator} from '../ErrorIndicator';

describe('ErrorIndicator', () => {
    const wrapper: ShallowWrapper = shallow(<ErrorIndicator />);

    it('should render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should contains 3 <span/>', () => {
        expect(wrapper.find('span').length).toEqual(3);
    });
});
