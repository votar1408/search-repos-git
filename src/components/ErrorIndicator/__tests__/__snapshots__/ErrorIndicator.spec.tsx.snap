// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorIndicator should render correctly 1`] = `
<div
  className="error-indicator"
>
  <span>
    OOOPS...
  </span>
  <span>
    something has gone wrong
  </span>
  <span>
    (but we already fixing it)
  </span>
</div>
`;
